### PR TITLE
feat: remove external dependencies

### PR DESCRIPTION
**Pull request by odsod**

### CODE REVIEW

File: sg/logger.go [hunk 1]
```diff
@@ -1,56 +1,32 @@
...
 }
 
-func (s *logSink) WithName(name string) logr.LogSink {
-	s.name = name
-	return s
+// Logger returns the log.Logger attached to the provided context.
+func Logger(ctx context.Context) *log.Logger {
+	logger := ctx.Value(loggerContextKey{}).(*log.Logger)
+	if logger == nil {
+		panic("context is missing a Logger")
```
Comment (gust1n) [inline sg/logger.go:29]: Would it make sense to provide a default named logger if it's missing, instead of panicking?

Comment (odsod) [reply sg/logger.go:29]: Hmm! Good point!

Comment (odsod) [reply sg/logger.go]: Updated

File: sg/logger.go [hunk 2]
```diff
@@ -1,56 +1,31 @@
...
-	panic("implement me")
+// WithLogger attaches a log.Logger to the provided context.
+func WithLogger(ctx context.Context, logger *log.Logger) context.Context {
+	return context.WithValue(ctx, loggerContextKey{}, logger)
 }
 
-func (s *logSink) WithName(name string) logr.LogSink {
-	s.name = name
-	return s
+// Logger returns the log.Logger attached to the provided context.
```
Comment (gust1n) [inline sg/logger.go:25]: Then I also think the comment should make that clear (a default logger will be returned if the context doesn't contain one)

Comment (odsod) [reply sg/logger.go]: Updated

**Comment by gust1n** [comment]
LGTM!

**Review by viktorvoltaire — APPROVED** [review]
Love it, let’s go! 🥳

**Pull request merged**